DigitalLearningSolutions.Web/Controllers/NotificationPreferencesController.cs

```diff
@@ -14,9 +25,20 @@ public IActionResult Index()
             }
 
             var adminId = User.GetCustomClaimAsInt(CustomClaimTypes.UserAdminId);
+            var adminNotifications = new List<NotificationPreference>();
+            if (adminId.HasValue)
+            {
+                adminNotifications = notificationPreferenceService.GetNotificationPreferencesForAdmin(adminId.Value).ToList();
+            }
+
             var delegateId = User.GetCustomClaimAsInt(CustomClaimTypes.LearnCandidateId);
+            var delegateNotifications = new List<NotificationPreference>();
+            if (delegateId.HasValue)
+            {
+                delegateNotifications = notificationPreferenceService.GetNotificationPreferencesForDelegate(delegateId.Value).ToList();
+            }
 
-            var model = new NotificationPreferencesViewModel(adminId, delegateId);
+            var model = new NotificationPreferencesViewModel(adminId, delegateId, adminNotifications, delegateNotifications);

```

Comment:
1. Does the ViewModel need the admin and delegateIds anymore? (as I think we can just check the notifications list size, which will be equivalent)
2. Could we move GetDelegateNotifications (and admin) into a service (maybe NotificationPreferenceService if the data service is renamed to "DataService"), which can accept null id values and converts the preferences to a list, just to keep the controller as logic free as possible? Alternatively, I'd be happy for the data service to do the null check on id and return an empty collection if that's the case. :)